metomi/isodatetime/data.py

```diff
@@ -859,15 +885,34 @@ def __init__(self, expanded_year_digits=0, year=None, month_of_year=None,
             self.time_zone.minutes = _int_caster(time_zone_minute,
                                                  "time_zone_minute")
         self.time_zone.unknown = self.truncated and has_unknown_tz
-        if not self.truncated:
-            # Reduced precision date - e.g. 1970 - assume Jan 1, etc.
-            if (self.month_of_year is None and self.week_of_year is None and
-                    self.day_of_year is None):
-                self.month_of_year = 1
-            if self.month_of_year is not None and self.day_of_month is None:
-                self.day_of_month = 1
-            if self.week_of_year is not None and self.day_of_week is None:
-                self.day_of_week = 1
+
+        specified_month = self.month_of_year or self.day_of_month
+        specified_week = self.week_of_year or self.day_of_week
```

Comment:
Thought it was strange the specified_month to be the day_of_month, but looks like these variables are for validating the boundaries only, so no issue here I believe.

Reply:
I could change them to `is_month_specified = any((self.month_of_year, self.day_of_month))` etc if that makes more sense

Reply:
I think that's fine :+1: these variables are used only in the code right after their declaration. So not too hard to see what they are intended for.